Cannot build clr after installing latest VS nightly

Just updated to VS 2019 Int Preview 16.8.0 Preview 3.0 \[30504.10.main\], and builds are failing on my local box. Repros in both the _master_ and the _release/5.0-rc2_ branches.

Error appears even after running `git clean -xdf`.

```txt
> build.cmd -s clr -c Release
... snip ...
    20>C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\winbase.h(9461,5): warning C5105: macro expansion producing 'defined' has undefined behavior (compiling source file C:\runtime\artifacts\obj\coreclr\Windows_NT.x64.Release\src\inc\idls_out\cordebug_i.c) [C:\runtime\artifacts\obj\coreclr\Windows_NT.x64.Release\src\inc\corguids_obj.vcxproj]
    20>C:\Program Files (x86)\Windows Kits\10\Include\10.0.19041.0\um\winbase.h(9461,5): error C2220: the following warning is treated as an error (compiling source file C:\runtime\artifacts\obj\coreclr\Windows_NT.x64.Release\src\inc\idls_out\cordebug_i.c) [C:\runtime\artifacts\obj\coreclr\Windows_NT.x64.Release\src\inc\corguids_obj.vcxproj]
```

More info on the warning: https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/c5105

--- a/eng/native/configurecompiler.cmake
+++ b/eng/native/configurecompiler.cmake
@@ -444,7 +444,8 @@ if (MSVC)
 
   # Disable Warnings:
   # 4291: Delete not defined for new, c++ exception may cause leak.
-  add_compile_options(/wd4291)
+  # 5105: Windows SDK headers use 'defined' operator in some macros
+  add_compile_options(/wd4291 /wd5105)
 
   # Treat Warnings as Errors:
   # 4007: 'main' : must be __cdecl.

